src/uu/mkfifo/src/mkfifo.rs

```diff
@@ -25,20 +25,26 @@ mod options {
 pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     let matches = uu_app().try_get_matches_from(args)?;
 
-    let mode = match matches.get_one::<String>(options::MODE) {
-        // if mode is passed, ignore umask
-        Some(m) => match usize::from_str_radix(m, 8) {
-            Ok(m) => m,
-            Err(e) => {
-                return Err(USimpleError::new(
-                    1,
-                    translate!("mkfifo-error-invalid-mode", "error" => e),
-                ));
+    let umask = uucore::mode::get_umask();
+    let mut mode = 0o666; // Default mode for FIFOs
+
+    if let Some(m) = matches.get_one::<String>(options::MODE) {
+        if m.chars().any(|c| c.is_ascii_digit()) {
+            mode = uucore::mode::parse_numeric(mode, m, false).map_err(|e| {
+                USimpleError::new(1, translate!("mkfifo-error-invalid-mode", "error" => e))
+            })?;
+        } else {
+            for item in m.split(',') {
+                mode = uucore::mode::parse_symbolic(mode, item, umask, false).map_err(|e| {
+                    USimpleError::new(1, translate!("mkfifo-error-invalid-mode", "error" => e))
+                })?;
             }
-        },
-        // Default value + umask if present
-        None => 0o666 & !(uucore::mode::get_umask() as usize),
-    };
+        }
+    } else {
+        mode &= !umask; // Apply umask if no mode is specified
+    }
+
+    let mode = mode as usize;
```

Comment:
Why do you convert back to usize? `fs::Permissions::from_mode` will be happier with the original u32 it seems, so you can remove the cast there.

Agree with dropping the mut though... but maybe that's a sign this chunk of code might be better off in a separate function?

Reply:
Great suggestion!